src/sortStudents.ts

```diff
@@ -1,16 +1,65 @@
 
 export interface Student {
-  // describe Student interface
+  name: string;
+  surname: string;
+  age: number;
+  married: boolean;
+  grades: number[];
+
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name = 'name',
+  Surname = 'surname',
+  Age = 'age',
+  Married = 'married',
+  AverageGrade = 'grades'
 }
 
-// create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
+
+export function
+sortStudents(students: Student[], sortBy: SortType, order: SortOrder)
+  : Student[] {
+  const dublicate = [...students];
+
+  function average(array: number[]): number {
+    const sum = array.reduce((prev, current) => prev + current);
+
+    return sum / array.length;
+  }
+
+  switch (sortBy) {
+    case SortType.Surname:
+    case SortType.Name:
+    case SortType.Married:
+      if (order === 'asc') {
+        dublicate
+          .sort((a, b) => String(a[sortBy]).localeCompare(String(b[sortBy])));
+      } else {
+        dublicate
+          .sort((a, b) => String(b[sortBy]).localeCompare(String(a[sortBy])));
+      }
+      break;
+
+    case SortType.Age:
+      if (order === 'asc') {
+        dublicate.sort((a, b) => a.age - b.age);
+      } else {
+        dublicate.sort((a, b) => b.age - a.age);
+      }
+      break;
+
+    case SortType.AverageGrade:
+      if (order === 'asc') {
+        dublicate.sort((a, b) => average(a.grades) - average(b.grades));
+      } else {
+        dublicate.sort((a, b) => average(b.grades) - average(a.grades));
```

Comment:
```suggestion
      return order === 'asc'
      ? dublicate.sort((a, b) => average(a.grades) - average(b.grades));
      : dublicate.sort((a, b) => average(b.grades) - average(a.grades));
```
That is not a mistake, but you also can use the [ternary operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Conditional_Operator)